Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -630,5 +630,47 @@ TEST_F(SerializerTest, EncodesEmptyDocument) {
   ExpectRoundTrip(key, empty_value, update_time, proto);
 }
 
+TEST_F(SerializerTest, EncodesNonEmptyDocument) {
+  DocumentKey key = DocumentKey::FromPathString("path/to/the/doc");
+  FieldValue fields = FieldValue::ObjectValueFromMap({
+      {"foo", FieldValue::StringValue("bar")},
+      {"two", FieldValue::IntegerValue(2)},
+      {"nested", FieldValue::ObjectValueFromMap({
+                     {"fourty-two", FieldValue::IntegerValue(42)},
+                 })},
+  });
+  SnapshotVersion update_time = SnapshotVersion{{1234, 5678}};
+
+  google::firestore::v1beta1::Value inner_proto;
```

Comment:
Optional: this is a pretty long namespace name, perhaps introduce an alias?
```cpp
namespace v1beta1 = google::firestore::v1beta1; // Strawman name
// ...
v1beta1::Value inner_proto; // Still calls out the different namespace, but less verbose
```

Reply:
I'm mildly in favor of some alias, just because I find the verbose names hard to read. Another alias could be applied to `google::protobuf` (e.g., just `protobuf`, `gprotobuf`, `g_protobuf`, etc.). If you don't feel like doing this, that's fine with me too.

Reply:
Hm. I don't think the "?" in my last comment was clear. To elaborate; do you think that `v1beta1::` is sufficient to indicate that this is a libproto "thing" rather than something else? (nanopb also uses that, but it's spelled with underscores and can't be shortened.)

Reply:
I personally think it's fine, and it's easy to find the alias in this file if need be. But even `proto_v1beta` (or similar, e.g. `libproto_v1beta1`, etc.) is shorter than the full `google::firestore::v1beta1` -- I especially don't like the first two segments, but it's also not really clear that this namespace has anything to do with protos, despite all the verbosity.

Reply:
> I especially don't like the first two segments, but it's also not really clear that this namespace has anything to do with protos, despite all the verbosity.

Ha; good point. It just looks so libprotobuf-ish to me that I didn't consider that. :)

I don't mind `v1beta1_protos` (or `libproto_v1beta1`). I'll switch to `v1beta1_protos`, but not in this PR (since it'll touch just about everything in this file) and will instead issue a followup.

Reply:
Sounds good (as long as the other reviewers are on board with this, because like you said it's an extensive change).

Reply:
I don't have any strong preference on this, but I'm also okay with just doing `v1beta1::`.